app_dart/lib/src/service/firestore.dart

```diff
@@ -67,24 +72,88 @@ class FirestoreService {
     return databasesDocumentsResource.commit(commitRequest, kDatabase);
   }
 
-  Future<List<firestore.Task>> queryCommitTasks(String commitSha) async {
+  /// Returns all tasks running against the speificed [commitSha].
+  Future<List<Task>> queryCommitTasks(String commitSha) async {
+    final Map<String, Object> filterMap = <String, Object>{
+      '$kTaskCommitShaField = ': commitSha,
+    };
+    final List<Document> documents = await query(kTaskCollectionId, filterMap);
+    return documents.map((Document document) => Task.fromDocument(taskDocument: document)).toList();
+  }
+
+  /// Returns Firestore [Value] based on corresponding object type.
+  Value getValueFromFilter(Object comparisonOject) {
+    if (comparisonOject is int) {
+      return Value(integerValue: comparisonOject.toString());
```

Comment:
Is this meant to be toString()? Seems odd that you would need to supply the integer value as a string but boolean as it normally appears.

Reply:
Ewww. thanks for clarifying.